contracts/src/systems/end_turn.cairo

```diff
@@ -21,6 +20,8 @@ mod end_turn_system {
                 outcome: game.outcome
             }
         );
+        game = get!(ctx.world, game_id, Game);
+        set!(ctx.world, Energy { game_id: game_id, player: ctx.origin, remaining: game.turn + 1 });
```

Comment:
hmm no turn 2 both player should have 1 energy, turn 4 both should have 2
```suggestion
        set!(ctx.world, Energy { game_id: game_id, player: ctx.origin, remaining: game.turn / 2_u128 });
```
Make sure that you change the turns into u128 to have a proper division

Reply:
Can you explain to me how this should work, maybe i misunderstood the process of turns. I thought it was:

Turn 0 =>
- Player 1
    - Start with 1 energy
    - End of turn increment remaining energy by 1. Energy.remaining should not be equal to 2.
- Player 2
    - Start with 1 energy
    - End of turn increment remaining energy by 1. Energy.remaining should not be equal to 2.

Turn 1 =>
- Player 1
    - Start new turn with 2 energy
    - End of turn increment by remaining energy 1. Energy.remaining should not be equal to 3.
- Player 2
    - Start new turn with 2 energy
    - End of turn increment by remaining energy 1. Energy.remaining should not be equal to 3.

Reply:
ooooh i see what's happening, each time a player presses end_turn, the turn number is incremented. So at turn 8 for example each player will have played 4 turns and should have 4 energy. Does that solve the confusion ?